src/NumberField/NumberField.js

```diff
@@ -32,25 +52,34 @@ export default function NumberField({
     const newEventValue = { target: { value: newValue } };
     const updatedEvent = { ...event, ...newEventValue };
 
-    // if the value is less than minimum, set error
-    if (newValue < min && min !== undefined) {
-      setValueError(true);
-      setErrorMessage(`Value must be greater than or equal to ${min}`);
-      return;
+    // set the number
+    setNumber(event.target.value);
+
+    // is min satisfied
+    let minSatisfied = true;
+    if (min && newValue < min) {
+      minSatisfied = false;
     }
 
-    // if the value is greater than maximum, set error
-    if (newValue > max && max !== undefined) {
-      setValueError(true);
-      setErrorMessage(`Value must be less than or equal to ${max}`);
-      return;
+    // is max satisfied
+    let maxSatisfied = true;
+    if (max && newValue > max) {
```

Comment:
Your logic here won't work when max value is set to 0 because 0 !== true so your if statement will always be skipped. You can test this by setting max to 0 and min to some negative number and then type in a value greater than 0 and you won't get error message. You should also check the same can't occur in your min logic on line 60.